### PR TITLE
Resolving missing module

### LINKED ISSUE #164: Installation fails
My installation failed with the following message:
<pre>An error has occurred. See the log file
/home/sven/Ericsson/CodeCheckerEclipsePlugin/tests/org.codechecker.eclipse.rcp.it.tests/target/work/data/.metadata/.log.
[<font color="#729FCF"><b>INFO</b></font>] <b>------------------------------------------------------------------------</b>
[<font color="#729FCF"><b>INFO</b></font>] <b>Reactor Summary:</b>
[<font color="#729FCF"><b>INFO</b></font>] 
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.root 1.0.0-SNAPSHOT ........ <font color="#8AE234"><b>SUCCESS</b></font> [ 10.544 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.bundles 1.0.0-SNAPSHOT ..... <font color="#8AE234"><b>SUCCESS</b></font> [  0.797 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.plugin 0.0.6-SNAPSHOT ...... <font color="#8AE234"><b>SUCCESS</b></font> [ 12.096 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.features 1.0.0-SNAPSHOT .... <font color="#8AE234"><b>SUCCESS</b></font> [  0.585 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.feature 1.0.0-SNAPSHOT ..... <font color="#8AE234"><b>SUCCESS</b></font> [  1.072 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.releng 1.0.0-SNAPSHOT ...... <font color="#8AE234"><b>SUCCESS</b></font> [  0.730 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.update 1.0.0-SNAPSHOT ...... <font color="#8AE234"><b>SUCCESS</b></font> [  3.434 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.target 1.0.0-SNAPSHOT ...... <font color="#8AE234"><b>SUCCESS</b></font> [  0.592 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.tests 1.0.0-SNAPSHOT ....... <font color="#8AE234"><b>SUCCESS</b></font> [  0.434 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.rcp.shared 1.0.0-SNAPSHOT .. <font color="#8AE234"><b>SUCCESS</b></font> [  0.854 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.rcp.unit.tests 1.0.0-SNAPSHOT <font color="#8AE234"><b>SUCCESS</b></font> [  7.579 s]
[<font color="#729FCF"><b>INFO</b></font>] org.codechecker.eclipse.rcp.it.tests 1.0.0-SNAPSHOT  <font color="#EF2929"><b>FAILURE</b></font> [ 12.011 s]
[<font color="#729FCF"><b>INFO</b></font>] <b>------------------------------------------------------------------------</b>
[<font color="#729FCF"><b>INFO</b></font>] <font color="#EF2929"><b>BUILD FAILURE</b></font>
[<font color="#729FCF"><b>INFO</b></font>] <b>------------------------------------------------------------------------</b>
[<font color="#729FCF"><b>INFO</b></font>] Total time:  03:54 min
[<font color="#729FCF"><b>INFO</b></font>] Finished at: 2019-07-02T13:18:42+02:00
[<font color="#729FCF"><b>INFO</b></font>] <b>------------------------------------------------------------------------</b>
[<font color="#EF2929"><b>ERROR</b></font>] Failed to execute goal <font color="#4E9A06">org.eclipse.tycho:tycho-surefire-plugin:1.3.0:test</font> <b>(default-test)</b> on project <font color="#06989A">org.codechecker.eclipse.rcp.it.tests</font>: <font color="#EF2929"><b>An unexpected error occured while launching the test runtime (return code 13). See log /home/sven/Ericsson/CodeCheckerEclipsePlugin/tests/org.codechecker.eclipse.rcp.it.tests/target/work/data/.metadata/.log for details.</b></font> -&gt; <b>[Help 1]</b>
[<font color="#EF2929"><b>ERROR</b></font>] 
[<font color="#EF2929"><b>ERROR</b></font>] To see the full stack trace of the errors, re-run Maven with the <b>-e</b> switch.
[<font color="#EF2929"><b>ERROR</b></font>] Re-run Maven using the <b>-X</b> switch to enable full debug logging.
[<font color="#EF2929"><b>ERROR</b></font>] 
[<font color="#EF2929"><b>ERROR</b></font>] For more information about the errors and possible solutions, please read the following articles:
[<font color="#EF2929"><b>ERROR</b></font>] <b>[Help 1]</b> http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[<font color="#EF2929"><b>ERROR</b></font>] 
[<font color="#EF2929"><b>ERROR</b></font>] After correcting the problems, you can resume the build with the command
[<font color="#EF2929"><b>ERROR</b></font>]   <b>mvn &lt;goals&gt; -rf :org.codechecker.eclipse.rcp.it.tests</b>
</pre>

The link to apache.org gave no useful information.
The log-file is attached.
[LogFile.txt](https://github.com/Ericsson/CodeCheckerEclipsePlugin/files/3349881/LogFile.txt)
Any help is welcome.

### PR DESCRIPTION
**Pull request by Vodorok**
!!!
Bumped minimal required eclipse to 4.8 Photon
!!!

This from this eclipse version a newer java support is enabled.
See more at: https://projects.eclipse.org/projects/eclipse/releases/4.8.0

Bumped Tycho to 1.4.0.
Added Javax.annotation bundle to target platform.
Added vmargs to the integration test pom.xml.
Added profiles to jdk 8, 9, 10 & 11.

Fix for the missing dependency with java 11 in the integration test phase.
Referenced in:
* https://wiki.eclipse.org/Tycho/Release_Notes/1.4
* https://bugs.eclipse.org/bugs/show_bug.cgi?id=541403

Resolves #164 
And depends on #166

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -8,9 +8,9 @@ This is a C/C++ code analysis plugin for Eclipse that shows bugs detected by the
 
 * Linux operating system
 * Recent [CodeChecker](https://github.com/Ericsson/codechecker) (6.8.0 and up)
-* Every eclipse version since [Oxygen 1](https://www.eclipse.org/downloads/packages/release/oxygen/1) (v4.7.0) is supported
+* Every eclipse version since [Phtoton](https://www.eclipse.org/downloads/packages/release/photon/r) (v4.8.0) is supported
```
Comment (gyorb) [inline readme.md]: I think we should split up the documentation to be more explicit what are the plugin build requirements and what is the runtime requirement for it.

This change mainly effects the build environment if I'm right.

Comment (Vodorok) [reply readme.md]: True that! 
But for the current setup of the compilation, we only test for one target platform, so until we create a deeper matrix with separate, older (newer) target platforms, we can't be sure the plugin works on older releases, even though we won't use newer features. I'll create an issue regarding this problem.
Until then, I can extend the readme with a note that it should work on Oxygen.

File: tests/org.codechecker.eclipse.rcp.it.tests/pom.xml [hunk 2]
```diff
@@ -13,37 +13,177 @@
...
-                    <useUIHarness>true</useUIHarness>
-                    <useUIThread>false</useUIThread>
-                    <product>org.codechecker.eclipse.feature</product>
-                    <application>org.eclipse.ui.ide.workbench</application>
-                    <testClass>org.codechecker.eclipse.plugin.AllTests</testClass>
-                </configuration>
-            </plugin>
-        </plugins>
-    </build>
+    <profiles>
```
Comment (Vodorok) [inline tests/org.codechecker.eclipse.rcp.it.tests/pom.xml]: We have to use different configurations depending on the jdk verisons. Maven profiles can only alter build segments, thats why this segment got a little bit verbose.

**Comment by Vodorok** [comment]
Resolves the first part of #132 (update the currently used Tycho version to a newer (latest?))

**Pull request merged**